src/cudadev/CUDADataFormats/TrackingRecHit2DSOAView.h

```diff
@@ -22,40 +24,84 @@ class TrackingRecHit2DSOAView {
-  __device__ __forceinline__ uint16_t detectorIndex(int i) const { return __ldg(m_detInd + i); }
-
+  // Our arrays do not require specific alignment as access will not be coalesced in the current implementation
+  // Sill, we need the 32 bits integers to be aligned, so we simply declare the SoA with the 32 bits fields first
+  // and the 16 bits behind (as they have a looser alignment requirement. Then the SoA can be create with a byte 
+  // alignment of 1)
+  generate_SoA_store(HitsStore,
+    // 32 bits section
+    // local coord
+    SoA_column(float, xLocal),
+    SoA_column(float, yLocal),
+    SoA_column(float, xerrLocal),
+    SoA_column(float, yerrLocal),
+    
+    // global coord
+    SoA_column(float, xGlobal),
+    SoA_column(float, yGlobal),
+    SoA_column(float, zGlobal),
+    SoA_column(float, rGlobal),
+    // global coordinates continue in the 16 bits section
+
+    // cluster properties
+    SoA_column(int32_t, charge),
+          
+    // 16 bits section (and cluster properties immediately continued)
+    SoA_column(int16_t, clusterSizeX),
+    SoA_column(int16_t, clusterSizeY)
+  );
+  
+  generate_SoA_store(SupportObjectsStore,
+    // This is the end of the data which is transferred to host. The following columns are supporting 
+    // objects, not transmitted 
+    
+    // Supporting data (32 bits aligned)
+    SoA_column(TrackingRecHit2DSOAStore::PhiBinner::index_type, phiBinnerStorage),
+          
+    // global coordinates (not transmitted)
+    SoA_column(int16_t, iphi),
+          
+    // cluster properties (not transmitted)
+    SoA_column(uint16_t, detectorIndex)
+  );
+  
+  generate_SoA_view(HitsAndSupportView,
+    SoA_view_store_list(
+      SoA_view_store(HitsStore, hitsStore),
+      SoA_view_store(SupportObjectsStore, supportObjectsStore)
+    ),
+    SoA_view_value_list(
+      SoA_view_value(hitsStore, xLocal, xLocal),
+      SoA_view_value(hitsStore, yLocal, yLocal),
+      SoA_view_value(hitsStore, xerrLocal, xerrLocal),
+      SoA_view_value(hitsStore, yerrLocal, yerrLocal),
+      
+      SoA_view_value(hitsStore, xGlobal, xGlobal),
+      SoA_view_value(hitsStore, yGlobal, yGlobal),
+      SoA_view_value(hitsStore, zGlobal, zGlobal),
+      SoA_view_value(hitsStore, rGlobal, rGlobal),
+      
+      SoA_view_value(hitsStore, charge, charge),
+      SoA_view_value(hitsStore, clusterSizeX, clusterSizeX),
+      SoA_view_value(hitsStore, clusterSizeY, clusterSizeY),
+      
+      SoA_view_value(supportObjectsStore, phiBinnerStorage, phiBinnerStorage),
+      SoA_view_value(supportObjectsStore, iphi, iphi),
+      SoA_view_value(supportObjectsStore, detectorIndex, detectorIndex)
+    )
+  );
+  
+  // Shortcut operator saving the explicit calls to view in usage.
+  __device__ __forceinline__ HitsAndSupportView::element operator[] (size_t index) { 
+    return m_hitsAndSupportView[index]; 
+  }
+  __device__ __forceinline__ const HitsAndSupportView::const_element operator[] (size_t index) const { 
```

Comment:
```suggestion
  __device__ __forceinline__ HitsAndSupportView::const_element operator[] (size_t index) const { 
```
?

Reply:
Yes, that was quite redundant. I fixed it.